consensus/hotstuff/votecollector/statemachine.go

```diff
@@ -199,7 +224,8 @@ func (m *VoteCollector) ProcessBlock(proposal *model.Proposal) error {
 		case hotstuff.VoteCollectorStatusInvalid: /* no op */
 
 		default:
-			return fmt.Errorf("VoteProcessor reported unknown status %s", proc.Status())
+			return fmt.Errorf("VoteProcessor reported unknown status %s when processing block %v", proc.Status(),
+				proposal.Block.BlockID)
```

Comment:
same here: the status of the vote processor is independent of the block that is currently being processed

```suggestion
			return fmt.Errorf("while processing block %v, found that VoteProcessor reported unknown status %s", proposal.Block.BlockID, proc.Status())
```